src/transformers/trainer.py

```diff
@@ -748,18 +757,18 @@ def _wrap_model(self, model, training=True):
             return model
 
         # Distributed training (should be after apex fp16 initialization)
-        if self.sharded_dpp:
+        if self.sharded_ddp is not None:
             # Sharded DDP!
-            if self.args.sharded_ddp == ShardedDDPType.SIMPLE:
+            if self.sharded_ddp == ShardedDDPOption.SIMPLE:
                 model = ShardedDDP(model, self.optimizer)
             else:
                 mixed_precision = self.args.fp16
-                cpu_offload = self.args.sharded_ddp in [ShardedDDPType.ZERO_2_OFFLOAD, ShardedDDPType.ZERO_3_OFFLOAD]
-                zero_3 = self.args.sharded_ddp in [ShardedDDPType.ZERO_3, ShardedDDPType.ZERO_3_OFFLOAD]
+                cpu_offload = ShardedDDPOption.OFFLOAD in self.args.sharded_ddp
+                zero_3 = self.sharded_ddp == ShardedDDPOption.ZERO_DP_3
                 # Breaking the self.model convention but I see no way around it for now.
```

Comment:
```suggestion
                # XXX: Breaking the self.model convention but I see no way around it for now.
```

Perhaps let's add a flag to get back to it later? Not attached it - this is usually how I know to come back to something later.